pilot/pkg/config/memory/store.go

```diff
@@ -194,56 +191,18 @@ func (cr *store) Update(cfg config.Config) (string, error) {
 		return "", errNotFound
 	}
 
-	_, exists = ns.Load(cfg.Name)
-	if !exists {
-		return "", errNotFound
-	}
-
-	existing, exists := ns.Load(cfg.Name)
+	oldCfg, exists := ns.Load(cfg.Name)
 	if !exists {
 		return "", errNotFound
 	}
-	if hasConflict(existing.(config.Config), cfg) {
-		return "", errConflict
+	if cfg.ResourceVersion == "" || oldCfg.(config.Config).ResourceVersion != cfg.ResourceVersion {
```

Comment:
Why remove `hasConflict` which is self documenting with a more complex boolean?

Reply:
https://github.com/istio/istio/blob/ce3ba502930b8cc7fa8c41ddcf1f8c3c8d9debc0/pilot/pkg/config/memory/store.go#L289
Set the resourceVersion to the current time when creating cfg. If resourceVersion is not empty it will always return true when executed in this place.

Reply:
> Why remove `hasConflict` which is self documenting with a more complex boolean?

Modified a bit